rest/src/plugins/cmc/cmcRoutes.js

```diff
@@ -42,42 +42,42 @@ module.exports = {
 				.then(fileData => ini.parse(fileData));
 		};
 
-		const lookupMosaicAmount = (mosaics, currencyId) => {
+		const lookupMosaicAmount = (mosaics, currencyMosaicId) => {
 			const matchingMosaic = mosaics.find(mosaic => {
 				const mosaicId = longToUint64(mosaic.id); // convert Long to uint64
-				return 0 === uint64.compare(currencyId, mosaicId);
+				return 0 === uint64.compare(currencyMosaicId, mosaicId);
 			});
 
 			return undefined === matchingMosaic ? 0 : parseInt(matchingMosaic.amount.toString(), 10);
 		};
 
 		server.get('/network/currency/supply/circulating', (req, res, next) => readAndParseNetworkPropertiesFile()
 			.then(async propertiesObject => {
-				const currencyId = propertyValueToMosaicId(propertiesObject.chain.currencyMosaicId);
-				const mosaics = await db.mosaicsByIds([currencyId]);
+				const currencyMosaicId = propertyValueToMosaicId(propertiesObject.chain.currencyMosaicId);
+				const mosaics = await db.mosaicsByIds([currencyMosaicId]);
 				const accounts = await db.catapultDb.accountsByIds([
 					{
 						[AccountType.publicKey]: propertiesObject.network.nemesisSignerPublicKey
 					}
 				]);
 
 				const totalSupply = parseInt(mosaics[0].mosaic.supply.toString(), 10);
-				const burnedSupply = parseInt(lookupMosaicAmount(accounts[0].account.mosaics, currencyId).toString(), 10);
+				const burnedSupply = lookupMosaicAmount(accounts[0].account.mosaics, currencyMosaicId);
 				sender.sendPlainText(res, next)(cmcUtils.convertToRelative(totalSupply - burnedSupply));
 			}));
 
 		server.get('/network/currency/supply/total', (req, res, next) => readAndParseNetworkPropertiesFile()
 			.then(propertiesObject => {
-				const currencyId = propertyValueToMosaicId(propertiesObject.chain.currencyMosaicId);
-				return db.mosaicsByIds([currencyId]).then(response => {
-					const supply = response[0].mosaic.supply.toString();
+				const currencyMosaicId = propertyValueToMosaicId(propertiesObject.chain.currencyMosaicId);
+				return db.mosaicsByIds([currencyMosaicId]).then(response => {
+					const supply = parseInt(response[0].mosaic.supply.toString(), 10);
```

Comment:
why is there .toString and parseInt needed here? o0
`mosaic.supply` here is hex string or something else?

Reply:
it is MongoDb.Long

Reply:
although that's from an old revision, and that should be removed